### PR TITLE
Fix macOS High Sierra build by moving to a new .NET Core SDK and disabling legacy tests

### PR DESCRIPTION
**Pull request by DustinCampbell**
This is getting really horrible now. In order to allow us to build on High Sierra, we need a newer version of the .NET Core SDK. However, that newer version uses newer NuGet packages, which results in our tests failing unless we update to newer NuGet packages in OmniSharp. And, to make matters more complicated, those newer NuGet packages are incompatible with the obsolete Microsoft.DotNet.ProjectModel package that we use to drive OmniSharp.DotNet and OmniSharp.Scripting.

I just built this on Windows and tested with a simple project.json project and it worked. However, I'm not sure how well it works.

**Edit**: I've decided to just bring the problematic code into OmniSharp itself. Now, `Microsoft.DotNet.ProjectModel` is `OmniSharp.DotNet.ProjectModel`, which will allow us to control how and when we decide to sunset project.json support.

### CODE REVIEW

**Comment by DustinCampbell** [comment]
got some failing tests to look at

**Comment by DustinCampbell** [comment]
Looks like the failing tests are all vstest issues. Hmmm...

**Comment by DustinCampbell** [comment]
fingers crossed!

**Comment by DustinCampbell** [comment]
@david-driscoll, @filipw - please take a look when you get a chance. I might look daunting, but the vast majority of the PR is just pulling in the code from https://github.com/dotnet/cli/tree/rel/1.0.0-preview2.1/src/Microsoft.DotNet.ProjectModel.

File: .travis.yml [hunk 1]
```diff
@@ -5,11 +5,11 @@ env:
   global:
     secure: m2PtYwYOhaK0uFMZ19ZxApZwWZeAIq1dS//jx/5I3txpIWD+TfycQMAWYxycFJ/GJkeVF29P4Zz1uyS2XKKjPJpp2Pds98FNQyDv3OftpLAVa0drsjfhurVlBmSdrV7GH6ncKfvhd+h7KVK5vbZc+NeR4dH7eNvN/jraS//AMJg=
 mono: latest
-dotnet: 1.0.4
+dotnet: 2.0.0
 os:
   - linux
   - osx
-osx_image: xcode7.3
+osx_image: xcode8.1
```
Comment (DustinCampbell) [inline .travis.yml]: Note: This makes Travis use OSX Sierra, which is required by .NET Core SDK 2.0.

File: build.cake [hunk 2]
```diff
@@ -51,52 +97,7 @@ Task("Setup")
     .IsDependentOn("InstallMonoAssets")
     .IsDependentOn("CreateMSBuildFolder");
 
-void ParseDotNetInfoValues(IEnumerable<string> lines, out string version, out string rid, out string basePath)
```
Comment (DustinCampbell) [inline build.cake]: This is no longer needed since the platform-sniffing code has been improved

File: build.json [hunk 3]
```diff
@@ -1,7 +1,10 @@
 {
-  "DotNetInstallScriptURL": "https://raw.githubusercontent.com/dotnet/cli/rel/1.0.0/scripts/obtain",
+  "DotNetInstallScriptURL": "https://raw.githubusercontent.com/dotnet/cli/release/2.0.0/scripts/obtain",
   "DotNetChannel": "preview",
-  "DotNetVersion": "1.0.4",
+  "DotNetVersion": "2.0.0",
+  "DotNetSharedRuntimeVersions": [
+    "1.1.2"
```
Comment (DustinCampbell) [inline build.json]: ℹ️ .NET Core SDK 2.0 does not install with a shared runtime for 1.1. However, several of the projects in our test target netcoreapp1.1. So, I've opted to just install the additional shared runtimes with our local .NET Core SDK so these projects can restore and build properly.

File: build/Packages.props [hunk 4]
```diff
@@ -7,18 +7,16 @@
...
+    <MicrosoftBuildVersion>15.3.409</MicrosoftBuildVersion>
+    <MicrosoftBuildFrameworkVersion>15.3.409</MicrosoftBuildFrameworkVersion>
+    <MicrosoftBuildTasksCoreVersion>15.3.409</MicrosoftBuildTasksCoreVersion>
+    <MicrosoftBuildUtilitiesCoreVersion>15.3.409</MicrosoftBuildUtilitiesCoreVersion>
+    <MicrosoftCodeAnalysisCommonVersion>2.3.2</MicrosoftCodeAnalysisCommonVersion>
+    <MicrosoftCodeAnalysisCSharpVersion>2.3.2</MicrosoftCodeAnalysisCSharpVersion>
+    <MicrosoftCodeAnalysisCSharpFeaturesVersion>2.3.2</MicrosoftCodeAnalysisCSharpFeaturesVersion>
+    <MicrosoftCodeAnalysisCSharpScriptingVersion>2.3.2</MicrosoftCodeAnalysisCSharpScriptingVersion>
+    <MicrosoftCodeAnalysisCSharpWorkspacesVersion>2.3.2</MicrosoftCodeAnalysisCSharpWorkspacesVersion>
+    <MicrosoftCodeAnalysisWorkspacesCommonVersion>2.3.2</MicrosoftCodeAnalysisWorkspacesCommonVersion>
```
Comment (DustinCampbell) [inline build/Packages.props]: ℹ️ I'm taking the opportunity to move a few unrelated packages forward (like Roslyn & MSBuild) as part of this change.

File: src/OmniSharp.DotNet.ProjectModel/AnalyzerOptions.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
```
Comment (DustinCampbell) [inline src/OmniSharp.DotNet.ProjectModel/AnalyzerOptions.cs]: ℹ️ Per the MIT license, we need to keep this copyright header. I've also included the LICENSE.md from https://github.com/dotnet/cli in the root of OmniSharp.DotNet.ProjectModel.

**Comment by DustinCampbell** [comment]
Hmmm... something about my changes to the legacy test projects causes tests to hang.

**Comment by david-driscoll** [comment]
I'll bring up project.json support in the next meeting.. which I'm going to schedule now as I sit at the Microsoft office in Charlotte...

**Comment by DustinCampbell** [comment]
Sounds good @david-driscoll

**Comment by DustinCampbell** [comment]
OK. I think this is much more stable now. Whew!

**Pull request merged**